WikiWatershed/model-my-watershed
# Add API and Password Management Page

## Overview

Fill in Account Page -> Account:

- Add section to copy and regenerate your API key
- Add button to reset password
    - Add user modal view to reset password
    - Add new django `change_password` view

See wireframes in original issue. 

See commit messages for more details. 

Connects #2188 
### Demo

![et31bcfymy](https://user-images.githubusercontent.com/7633670/31553479-a0394bf2-b008-11e7-83ef-227f65d28a1d.gif)

Error and loading states for API key (throttling turned on to make it obvious)
![nco1mlrrub](https://user-images.githubusercontent.com/7633670/31553484-a2443e3e-b008-11e7-887e-6a03be686839.gif)

Error states, reset password....still pretty tragically styled 
<img width="521" alt="screen shot 2017-10-13 at 11 21 59 am" src="https://user-images.githubusercontent.com/7633670/31553537-cdc0de6e-b008-11e7-989d-f6d6f540b28b.png">
<img width="490" alt="screen shot 2017-10-13 at 11 21 45 am" src="https://user-images.githubusercontent.com/7633670/31553538-cdc953dc-b008-11e7-916a-00f03fa8bc8b.png">


## Testing Instructions

 * Pull and `bundle`
 * http://localhost:8000/account --> Account
 * Confirm you can get the api key sin your clipboard by clicking the "copy" button
* Confirm the "regenerate key" button updates the view with a new key (and displays a warning
modal)
* Confirm the reset password button opens a modal that allows resetting your password
* Confirm making your browser offline shows an error message when you try to regenerate your api key
